### PR TITLE
Extend node-style-guide base

### LINKED ISSUE #5: Extend node-style-guide?
We have [**node-style-guide**](https://www.npmjs.com/package/node-style-guide) and [**eslint-config-node-style-guide**](https://github.com/pdehaan/eslint-config-node-style-guide/blob/master/index.js) to choose from.

But it may be good to use **node-style-guide** as a base set of rules and then define our overrides in **eslint-config-gulp**. It'd make it easier to track that we've overridden the `"object-curly-spacing": [2, "always"]` rule (or any other future rules).

Basically, `npm i eslint-config-node-style-guide -D` and then set our eslint.json like:

``` js
{
  "extends": "node-style-guide",
  "rules": {
    "object-curly-spacing": [2, "always"]
  }
}
```

### PR DESCRIPTION
**Pull request by pdehaan**
Closes #5

### CODE REVIEW

**Comment by phated** [comment]
Awesome! Glad you got added as contrib on node-style-guide

**Comment by phated** [comment]
@pdehaan this doesn't seem to work when I install it into the vinyl project.  Not sure what is wrong but it can't find "eslint-config-node-style-guide"

**Comment by pdehaan** [comment]
I think it's an ESLint oddity, you need to add the "eslint-config-node-style-guide" module to the repo also.

**Comment by phated** [comment]
@pdehaan it looks like we have to require it directly, as stated in https://github.com/eslint/eslint/issues/3458#issuecomment-149751603

**Comment by pdehaan** [comment]
That's the basic approach I went with in https://github.com/pdehaan/eslint-config-node-style-guide/blob/master/index.js

We could do something similar here, if you wanted to avoid having to add **eslint-config-node-style-guide** as a devDependency if each project.

Maybe something like changing "eslint.json" to "index.js" which has the following:

``` js
var cfg = require('node-style-guide').eslintrc;
cfg.rules['object-curly-spacing'] = [2, 'always'];
module.exports = cfg;
```

Not super pretty, but I guess avoids adding a dependency to each repo.
I can try looking at other custom, nested ESLint configs and see if they have any other/better approaches since I don't find my solution above to be exceptionally readable.

File: main.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+'use strict';
+
+var join = require('path').join;
+
+module.exports = {
+  extends: join(__dirname, '_base.js'),
```
Comment (pdehaan) [inline main.js:6]: Not sure why, but using "./_base.js" here didn't work for me.
Feels like an ESLint bug. I'll try to triage and submit a bug shortly.

**Comment by pdehaan** [comment]
Once we can get this landed and published (probably another major version bump), I'll try and submit PRs for the other repos.

File: main.js (hunk 1 above)
Comment (pdehaan) [inline main.js:6]: Filed as https://github.com/eslint/eslint/issues/4263

**Comment by phated** [comment]
@pdehaan it looks like the referenced issue has been fixed.  Are we still looking to merge something like this PR?

**Comment by pdehaan** [comment]
Ah, forgot all about this. Considering it's been 7-8 months, I'm guessing it's probably pretty safe to just kill this PR and revisit it in the future if we think it adds value. I don't think that the node style guide has seen many PRs, so probably no real harm in copy-pasting all the rules and keeping them duplicated here.

**Comment by phated** [comment]
👍  cool thanks.

**Pull request closed without merging**